radioactivedecay/utils.py

```diff
@@ -307,42 +376,47 @@ def parse_nuclide_str(nuclide: str) -> str:
 
     """
 
+    original_input = nuclide
     nuclide = "".join(nuclide.split())  # Remove all whitespaces (Issue #65).
+    nuclide = nuclide.replace("-", "", 1)  # Strip out first hyphen
+    if not nuclide.isalnum():
+        raise NuclideStrError(
+            original_input,
+            "Nuclide strings must contain only letters, numbers, and (optionally) up to one "
+            "hyphen.",
+        )
+
+    A = "".join([n for n in nuclide if n.isdigit()])  # Mass number
+    if len(A) == 0 or int(A) > 300:  # Largest mass number in NUBASE2020 is 295
+        raise NuclideStrError(original_input, f"Mass number ({A}) is unphysical.")
 
-    letter_flag, number_flag = False, False
-    for char in nuclide:
-        if char.isalpha():
-            letter_flag = True
-        if char.isdigit():
-            number_flag = True
-        if letter_flag and number_flag:
-            break
-
-    if not (letter_flag and number_flag) or len(nuclide) < 2 or len(nuclide) > 7:
-        raise ValueError(str(nuclide) + " is not a valid nuclide string.")
-
-    while nuclide[0].isdigit():  # Re-order inputs e.g. 99mTc to Tc99m.
-        nuclide = nuclide[1:] + nuclide[0]
-    if nuclide[0] in get_metastable_chars():
-        nuclide = nuclide[1:] + nuclide[0]
-
-    for idx in range(1, len(nuclide)):  # Add hyphen e.g. Tc99m to Tc-99m.
-        if nuclide[idx].isdigit():
-            if nuclide[idx - 1] != "-":
-                nuclide = f"{nuclide[:idx]}-{nuclide[idx:]}"
-            break
-
-    # Convert incorrect capitalizations, e.g. tc-99M to Tc-99m, tC-99m to Tc-99m (Issue #65).
-    # Note 99MTc will fail because above logic requires metastable state char is lower case.
-    if nuclide[0].islower() or nuclide[1].isupper() or nuclide[-1].isupper():
-        nuclide = (
-            nuclide[0].upper()
-            + nuclide[1].lower()
-            + nuclide[2:-1]
-            + nuclide[-1].lower()
+    alpha_components = nuclide.split(A)
+    if len(alpha_components) != 2:
+        raise NuclideStrError(original_input, "")
+
+    if alpha_components[0] == "":  # User inputted mass number first
+        metastable_char, element = _process_metastable_element_str(alpha_components[1])
+    else:  # User inputted element symbol first
+        element = alpha_components[0]
```

Comment:
Do we consider the case where the metastable state char is combined with the element symbol? I.e. Tcm99?

Reply:
No we don't. Again Tcm99 has a few hits on Google: https://www.google.com/search?q="Tcm99"

We have two options:
- Easy: just say we don't support the two formats you raised.
- Effort required: Add support for them.

I'd say lets take the easy option first. Later if someone comes with a use case which requires the other formats, we could always try adding them.

Reply:
Sounds good to me, I agree.